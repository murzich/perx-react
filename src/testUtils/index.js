// eslint-disable-next-line import/prefer-default-export
export { default as renderWithRedux } from './renderWithRedux';
