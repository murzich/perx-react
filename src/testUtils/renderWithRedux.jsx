// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer from '../state/reducer';

export default function renderWithRedux(
  ui,
  { initialState, store = createStore(reducer, initialState) } = {},
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
}

export const rerenderWithRedux = rerender => (
  ui,
  { initialState, store = createStore(reducer, initialState) } = {},
) => ({
  ...rerender(<Provider store={store}>{ui}</Provider>),
  store,
});
