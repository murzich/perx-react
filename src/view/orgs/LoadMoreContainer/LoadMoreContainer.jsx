import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import { orgsActions, orgsSelectors } from '../../../state';
import LoadMore from '../../shared/LoadMore';

const LoadMoreContainer = ({ nextPage, fetchNextPage }) => (
  <LoadMore fetchNextPage={fetchNextPage} nextPage={nextPage} />
);

LoadMoreContainer.propTypes = {
  nextPage: PropTypes.number,
  fetchNextPage: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  nextPage: orgsSelectors.getNextPage(state),
});

export default connect(
  mapStateToProps,
  { fetchNextPage: orgsActions.fetchMoreOrgsRequest },
)(LoadMoreContainer);
