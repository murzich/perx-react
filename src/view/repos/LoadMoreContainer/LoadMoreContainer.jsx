import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import { reposActions, reposSelectors } from '../../../state';
import LoadMore from '../../shared/LoadMore';

const LoadMoreContainer = ({ nextPage, fetchNextPage }) => (
  <LoadMore fetchNextPage={fetchNextPage} nextPage={nextPage} />
);

LoadMoreContainer.propTypes = {
  nextPage: PropTypes.number,
  fetchNextPage: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  nextPage: reposSelectors.getNextPage(state),
});

export default connect(
  mapStateToProps,
  { fetchNextPage: reposActions.fetchMoreReposRequest },
)(LoadMoreContainer);
