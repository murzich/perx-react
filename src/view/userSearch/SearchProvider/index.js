import SearchProvider from './SearchProvider';

export default SearchProvider;
