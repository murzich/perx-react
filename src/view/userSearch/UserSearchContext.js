import { createContext } from 'react';

const UserSearchContext = createContext();

export default UserSearchContext;
