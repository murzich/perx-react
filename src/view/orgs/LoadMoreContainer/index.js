import LoadMoreContainer from './LoadMoreContainer';

export default LoadMoreContainer;
