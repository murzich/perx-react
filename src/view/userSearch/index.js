import SearchForm from './SearchForm';
import SearchProvider from './SearchProvider';
import UserSearchContext from './UserSearchContext';

export { UserSearchContext, SearchProvider };

export default SearchForm;
