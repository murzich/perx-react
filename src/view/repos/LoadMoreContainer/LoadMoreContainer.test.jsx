import { cleanup, waitForElement } from '@testing-library/react';
import React from 'react';
import { renderWithRedux } from '../../../testUtils';
import { UserSearchContext } from '../../userSearch';
import LoadMoreContainer from './LoadMoreContainer';

describe('LoadMoreContainer Repos', () => {
  afterEach(cleanup);

  const testUser = 'testUser';

  it('should match snapshot with default props', () => {
    const { asFragment } = renderWithRedux(
      <UserSearchContext.Provider value={testUser}>
        <LoadMoreContainer />
      </UserSearchContext.Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot with store data', () => {
    const { asFragment } = renderWithRedux(
      <UserSearchContext.Provider value={testUser}>
        <LoadMoreContainer />
      </UserSearchContext.Provider>,
      {
        initialState: {
          repos: {
            nextPage: 2,
          },
        },
      },
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot with loading state', async () => {
    const { asFragment, getByText } = renderWithRedux(
      <UserSearchContext.Provider value={testUser}>
        <LoadMoreContainer />
      </UserSearchContext.Provider>,
      {
        initialState: {
          repos: {
            nextPage: 2,
            loadingMore: true,
          },
        },
      },
    );

    await waitForElement(() =>
      getByText('load more', {
        selector: '.loading',
      }),
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
