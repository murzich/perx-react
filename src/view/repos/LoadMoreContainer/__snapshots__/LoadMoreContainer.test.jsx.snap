// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadMoreContainer Repos should match snapshot with default props 1`] = `<DocumentFragment />`;

exports[`LoadMoreContainer Repos should match snapshot with loading state 1`] = `
<DocumentFragment>
  <button
    class="ui fluid loading disabled button"
    disabled=""
    tabindex="-1"
  >
    load more
  </button>
</DocumentFragment>
`;

exports[`LoadMoreContainer Repos should match snapshot with store data 1`] = `
<DocumentFragment>
  <button
    class="ui fluid button"
  >
    load more
  </button>
</DocumentFragment>
`;
